client/components/redirect-when-logged-in/index.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import debugFactory from 'debug';
+import { getCurrentUser } from 'state/current-user/selectors';
+
+const debug = debugFactory( 'calypso:redirect-when-logged-in' );
+
+class RedirectWhenLoggedIn extends React.Component {
+	static propTypes = {
+		delayAtMount: React.PropTypes.number,
+		redirectTo: React.PropTypes.string.isRequired,
+		replaceCurrentLocation: React.PropTypes.bool,
+		waitForEmailAddress: React.PropTypes.string,
+		waitForUserId: React.PropTypes.number,
+	};
+
+	doTheRedirect() {
+		debug( this.props.replaceCurrentLocation ? 'replace' : 'assign', this.props.redirectTo );
+		this.props.replaceCurrentLocation
+			? window.location.replace( this.props.redirectTo )
+			: window.location.assign( this.props.redirectTo );
```

Comment:
Assuming you don't want to use

```js
const methodName = foo ? 'replace' : 'assign';
window.location[ methodName ]( this.props.redirectTo );
```

in order to keep the code easily greppable, this is still a good place for Lodash's `invoke`:

```js
const methodPath = replaceCurrentLocation ? 'location.replace' : 'location.assign';
invoke( window, methodPath, redirectTo );
```